app/com/linkedin/drelephant/tuning/hbt/FitnessManagerHBT.java

```diff
@@ -204,8 +204,12 @@ protected void checkToDisableTuning(Set<JobDefinition> jobDefinitionSet) {
   private boolean disableTuningforHeuristicsPassed(JobDefinition jobDefinition,
       List<TuningJobExecutionParamSet> tuningJobExecutionParamSets, int numberOfAppliedSuggestedParamExecution) {
     //Minimum three execution needed for HBT to do some resource optimization
+    int validSuggestedParamExecutionCountAfterReEnable = TuningHelper.
+        getValidSuggestedParamExecutionCountAfterTuneInReEnable(jobDefinition.id, tuningJobExecutionParamSets);
     if (areHeuristicsPassed(tuningJobExecutionParamSets)
-        && numberOfAppliedSuggestedParamExecution >= MINIMUM_HBT_EXECUTION) {
+        && numberOfAppliedSuggestedParamExecution >= MINIMUM_HBT_EXECUTION &&
```

Comment:
Why do we need this condition now ?

numberOfAppliedSuggestedParamExecution >= MINIMUM_HBT_EXECUTION

Reply:
So that TuneIn has atleast N number of executions before disablement if the heuristics are passed. [Same reason as earlier]